packages/shared/src/utils/constants/contract.tsx

```diff
@@ -154,13 +154,19 @@ export const getContractTypesConfig: TGetContractTypesConfig = symbol => ({
         barrier_count: 1,
         components: ['trade_type_tabs', 'barrier_selector', 'take_profit'],
     },
-    vanilla: {
+    vanillalongcall: {
         title: localize('Call/Put'),
-        trade_types: ['VANILLALONGCALL', 'VANILLALONGPUT'],
+        trade_types: ['VANILLALONGCALL'],
+        basis: ['stake'],
+        components: ['duration', 'strike', 'amount', 'trade_type_tabs'],
+        barrier_count: 1,
+    },
+    vanillalongput: {
+        title: localize('Call/Put'),
+        trade_types: ['VANILLALONGPUT'],
         basis: ['stake'],
         components: ['duration', 'strike', 'amount', 'trade_type_tabs'],
         barrier_count: 1,
-        config: { should_override: true },
```

Comment:
I think after dividing `vanilla` into `vanillalongcall` and `vanillalongput` we don't need this field

Reply:
mmm id be hesitant to make this change here because of akmal's vanilla on financial change. it might affect his logic in displaying different descriptions and glossary. haha

Reply:
The reason of separation is that after switching between Call\Put multiple times they can broke. There is a video with example, pinned to the CU card, where after switching some errors from `Call` field validation were shown to `Put` and vice versa

Reply:
One more important thing, that was removed, is `vanilla_trade_type`. Now it's possible to use `contract_type` instead (like for the rest contract). `is_vanilla` is still working too of course.

Reply:
@kate-deriv if you're removing it, please also remove its usage in packages/trader/src/Stores/Modules/Trading/Helpers/contract-type.js

Reply:
@henry-deriv @akmal-deriv Please be advised 🙂
#### **The reason for the separation of `Call` & `Put` into 2 configs instead of 1** is based on the following. 🙂
**Current approach** in production for requesting & processing `proposal` response for vanilla `Call` & `Put` is the same as for `Rise/Fall`:
    - 1 config for both `Call` & `Put`,
    - 2 requests for both `Call` & `Put` are sent together when `Rise/Fall` or 1 of the vanillas is selected,
    - 2 proposal responses for `Call` & `Put` are processed in the same way.
#### **Why is this approach applicable to `Rise/Fall`?**
Because Rise/Fall is a single trade type for which both `Rise(Call)` & `Fall(Put)` `proposal` responses return the **same** data for the **same** trade parameters entered by the user.
####  **Why is this approach **not** applicable to vanilla `Call` & `Put`?**
Because `Call` & `Put` are 2 separate trade types for which both `vanilla Call` & `vanilla Put` `proposal` responses return **different** data for the **same** trade parameters entered by the user.
Current logic is causing issues because the same trade params, which work well for Call, cause errors for Put, and since currently this `proposal` response logic is united, 2 separate responses get mixed, and UI breaks here and there.
#### What is the new approach about?
For the reasons mentioned above, we are separating the logic for `vanilla Call` & `vanilla Put` in order to fix it once and for all instead of patching local issues here and there, in the same way like it's done for **Turbos**:
    - 2 configs for each vanilla type (Call & Put),
    - only 1 request is sent when one of the vanillas is selected,
    - each proposal response is processed separately.

Reply:
sounds good to me

Reply:
I don't think it's a good approach, if UI breaks, it should be handled over there, there was no mix in responses before, which means something else was broken at some point. What was the point to separate vanilla into call and put if it's the same trade type and there's no difference other than API response?

Reply:
@akmal-deriv Probably before, API responses returned the same values, maybe something changed on BE side.  I have to disagree that Call and Put are the same trade type. API response handling is the most important for the trade type. If API response is different, trade type has to be considered different, not the same. Also, if Call and Put were the same trade type, we wouldn't have to switch between them like we don't have to switch between Rise/Fall. :)